components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -354,7 +354,12 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
 
                 OAuthClientRequest authzRequest;
 
+                String scopes = getScope(authenticatorProperties);
+
                 String queryString = getQueryString(authenticatorProperties);
+                if (StringUtils.isNotBlank(scopes)) {
```

Comment:
What happens if the `queryString` already contains scopes? This will most likely be the case for migrated setups.

Reply:
If the query string already contains a scope, that value will be used. Scopes field will be given priority only if it is not blank. The priority is set as:
1. Scopes field value
2. Query params scope value
3. If both above are blank, use default openid

Reply:
If both queryString and scopes contain OIDC scopes, where do we ensure that the scopes will be given priority?